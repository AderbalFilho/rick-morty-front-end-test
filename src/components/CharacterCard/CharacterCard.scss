.character-card {
  position: relative;
}

.character-card--hoverable {
  cursor: pointer;

  &:hover {
    border: 2px solid var(--color-lime-green);
    box-shadow: var(--shadow-card-normal);
  }
}

.character-card__img {
  border-radius: var(--radius-medium);
  width: 220px;
}

.character-card__img--dead {
  filter: var(--filter-dead);
}

.character-card__text-container {
  background-color: var(--background-color-card-text);
  border-radius: var(--radius-character-text);
  bottom: 0;
  height: 55px;
  padding: 7px 12px;
  position: absolute;
  width: 100%;
}

.character-card__title {
  font-size: 1.25rem;
  line-height: 1.5625rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.character-card__subtitle {
  font-size: 0.75rem;
  line-height: 0.9375rem;
}
