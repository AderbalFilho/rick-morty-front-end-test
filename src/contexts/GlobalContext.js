import React, { createContext, useState } from 'react';

export const GlobalContext = createContext();

export const GlobalStorage = ({ children }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [characters, setCharacters] = useState([]);
  const [isLoading /*, setIsLoading*/] = useState(false);

  function handleSearch() {
    // Do search
    setCharacters([
      /* result */
    ]);
  }

  return (
    <GlobalContext.Provider
      value={{ searchTerm, setSearchTerm, characters, handleSearch, isLoading }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
