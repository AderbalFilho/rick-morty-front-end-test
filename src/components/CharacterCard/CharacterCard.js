import React, { useContext } from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import { GCard, GText, GTitle } from '@ui';
import { GlobalContext } from '@contexts/GlobalContext';

import './CharacterCard.scss';

function CharacterCard({ classes, classesImg, character, variant }) {
  const className = classNames(
    'character-card',
    variant === 'normal' && 'character-card--hoverable',
    classes
  );
  const classNameImg = classNames(
    'character-card__img',
    character.status === 'Dead' && 'character-card__img--dead',
    classesImg
  );
  const { setModalCharacter } = useContext(GlobalContext);

  function handleClick() {
    if (variant !== 'modal') {
      setModalCharacter(character);
    }
  }

  return (
    <GCard classes={className} onClick={handleClick}>
      <img
        src={character.image}
        alt={character.name}
        className={classNameImg}
      />
      <div className="character-card__text-container">
        <GTitle family="seravek" weight="bold" classes="character-card__title">
          {character.name}
        </GTitle>
        <GText family="seravek" classes="character-card__subtitle">
          {character.species}
        </GText>
      </div>
    </GCard>
  );
}

CharacterCard.defaultProps = {
  variant: 'normal',
};

CharacterCard.propTypes = {
  classes: PropTypes.string,
  classesImg: PropTypes.string,
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
  }).isRequired,
  variant: PropTypes.oneOf(['normal', 'modal']),
};

export default CharacterCard;
