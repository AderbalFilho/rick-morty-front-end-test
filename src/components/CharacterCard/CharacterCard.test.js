import React from 'react';
import TestRenderer from 'react-test-renderer';

import CharacterCard from './CharacterCard';
import { GlobalStorage } from '@contexts/GlobalContext';

describe('CharacterCard', () => {
  it('matches snapshot', () => {
    const character = {
      name: 'Rick Sanchez',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      species: 'Human',
      status: 'Alive',
    };
    const testRenderer = TestRenderer.create(
      <GlobalStorage>
        <CharacterCard character={character} />
      </GlobalStorage>
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
