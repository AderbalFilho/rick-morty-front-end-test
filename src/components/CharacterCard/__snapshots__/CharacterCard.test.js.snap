// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard matches snapshot 1`] = `
<div
  className="g-card g-card--normal character-card character-card--hoverable"
  onClick={[Function]}
>
  <img
    alt="Rick Sanchez"
    className="character-card__img"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <div
    className="character-card__text-container"
  >
    <h1
      className="g-title g-title--white g-title--seravek g-title--space-normal g-title--bold character-card__title"
    >
      Rick Sanchez
    </h1>
    <p
      className="g-text g-text--white g-text--seravek character-card__subtitle"
    >
      Human
    </p>
  </div>
</div>
`;
